File: app/src/main/java/reading/speed/improver/user/pupil/PupilsAdapter.java
package reading.speed.improver.user.pupil;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.TextView;
import androidx.annotation.NonNull;
import org.jetbrains.annotations.NotNull;
import reading.speed.improver.R;

import java.util.List;

public class PupilsAdapter extends ArrayAdapter<Pupil> {


    public PupilsAdapter(@NonNull Context context, int resource, @NonNull List<Pupil> objects) {
        super(context, resource, objects);
    }

    public interface PupilsAdapterListener {
        void onPupilClick(Pupil pupil);
    }

    private PupilsAdapterListener mListener;

    public void subscribeListener(PupilsAdapterListener pupilsAdapterListener) {
        mListener = pupilsAdapterListener;
    }


    @NotNull
    @Override
    public View getView(int position, View convertView, final ViewGroup parent) {

        if (convertView == null) {
            convertView = LayoutInflater.from(getContext()).inflate(R.layout.list_pupil, parent, false);
        }

        Pupil pupil = getItem(position);
        TextView textView = convertView.findViewById(R.id.pupil_textView);
        textView.setText(pupil.name);
        textView.setTag(position);
        textView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                int position = (Integer) view.getTag();
                Pupil pupil = getItem(position);
                mListener.onPupilClick(pupil);
            }
        });
        return convertView;
    }
}

File: app/src/main/java/reading/speed/improver/user/pupil/management/PupilsAdapter.kt
package reading.speed.improver.user.pupil.management

import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import org.jetbrains.anko.dip
import org.jetbrains.anko.linearLayout
import org.jetbrains.anko.padding
import org.jetbrains.anko.textView
import reading.speed.improver.repository.ChitaloidRepository
import reading.speed.improver.service.ChitaloidService
import reading.speed.improver.user.pupil.Pupil
import java.util.*

class PupilsAdapter(var list: ArrayList<Pupil> = ArrayList()) : BaseAdapter() {
    override fun getView(i: Int, v: View?, parent: ViewGroup?): View {
        return with(parent!!.context) {
            var taskNum: Int = i + 1
            linearLayout {
                textView {
                    text = list.get(i).name
                    textSize = 16f
                    padding = dip(10)
                }.lparams {
                    height = dip(50)
                }
            }
        }
    }

    override fun getItem(position: Int): Pupil {
        return list[position]
    }

    override fun getCount(): Int {
        return list.size
    }

    override fun getItemId(position: Int): Long {
        return 0L
    }

    fun getPupil(position: Int): Pupil {
        return list[position]
    }

    fun add(text: String) {
        val chitaloidService: ChitaloidService = ChitaloidService()
        var pupil: Pupil = chitaloidService.createPupil(text)
        chitaloidService.addPupil(pupil)
        pupil = chitaloidService.getPupilByName(pupil.name)
        list.add(list.size, pupil)
        notifyDataSetChanged()
    }

    fun delete(i: Int) {
        ChitaloidRepository.getInstance().deletePupil(list[i])
        list.removeAt(i)
        notifyDataSetChanged()
    }

}